### DISCUSSION TITLE
Spawn solc ENOENT: Type number and <Enter> or click with the mouse (q or empty cancels):

### DISCUSSION THREAD

**Discussion by MohammadRokib** [LunarVim/LunarVim#4345] [Q&A]
Facing this weird issue for some time while working with solidity files. Its shows the following error
```sh
Spawn solc ENOENT: Type number and <Enter> or click with the mouse (q or empty cancels):
```
Whenever I go into the insert mode this option shows at the bottom. No matter what I give here `q`, any number `<enter>` it goes away and comes back again after one click on the keyboard.

![Screenshot 2023-08-27 151225](https://github.com/LunarVim/LunarVim/assets/60141836/cadbaa86-40c6-478f-8455-741d9700bfb4)

**Comment by DragonDmoney** [comment]
An answer to this would be great!

**Comment by Danijel-Enoch** [comment]
Has this been solved?

Facing Same Issue

**Reply by DanEscher98** [reply]
Me too. Facing exactly the same problem

**Comment by soufianeamini** [comment]
Hi, I've faced this issue and came here and found nothing, now that I solved it I figured I'd drop an answer.
I've tried things such as downloading solcjs globally or redownloading the lsp but no cigar. What finally made it work was installing solc (which is not compatible with solcjs at all)

You can try downloading it from here: https://docs.soliditylang.org/en/latest/installing-solidity.html

As for what I personally did, since I'm using arch I've had to download the following aur packages:
https://aur.archlinux.org/packages/solidity
https://aur.archlinux.org/packages/evmone
The other dependencies can be found just fine using the official repository (such as cmake)

After that the lsp for solidity works just fine!

**Reply by arepaFlipper** [reply]
solc in nixos

**Comment by 0-x-joseph** [comment]
What solved this issue for me in ubuntu is just installing `solc` package from `apt`:
```bash
sudo apt update
sudo apt install solc
```

**Comment by bekicoder** [comment]
I am also facing the same problem in window 11 WSL. I am tried solving this problem😬😬

**Reply by bekicoder** [reply]
I fixed the problem by installing vscode-solidity-server solidity_ls   from nvim Mason  🎯🎯👍👍💕
